plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/ColumnIdentity.java

```diff
@@ -36,7 +37,8 @@
     private final int id;
     private final String name;
     private final TypeCategory typeCategory;
-    private final List<ColumnIdentity> children;
+    // LinkedHashMap is used to maintain the column ordering
+    private final LinkedHashMap<Integer, ColumnIdentity> children;
```

Comment:
is it possible to avoid encoding order here and rely on `RowType` field ordering wherever required?

Reply:
I don't think so, not with still doing

> Also, I think it's better to store iceberg ids as much as possible (e.g. instead of strings in path), for robustness.

[from this comment](https://github.com/trinodb/trino/pull/8129#discussion_r677002820)

The issue I ran into while trying to use Types is that `ColumnIdentity` doesn't have `Types`, only `TypeCategory`'s, making it difficult to step down the dereference chain that way.